lib/src/Pos/Logic/Types.hs

```diff
@@ -0,0 +1,207 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+module Pos.Logic.Types
+    ( LogicLayer (..)
+    , Logic (..)
+    , KeyVal (..)
+    , GetBlockError (..)
+    , GetBlockHeaderError (..)
+    , GetBlockHeadersError (..)
+    , GetTipError (..)
+    , dummyLogicLayer
+    ) where
+
+import           Universum
+import           Data.Default              (def)
+import           Data.Tagged               (Tagged)
+
+import           Pos.Communication         (NodeId, TxMsgContents)
+import           Pos.Core.Configuration    (HasGenesisBlockVersionData, genesisBlockVersionData)
+import           Pos.Core.Block            (Block, BlockHeader)
+import           Pos.Core.Types            (HeaderHash, StakeholderId,
+                                            ProxySKHeavy)
+import           Pos.Core.Txp              (TxId)
+import           Pos.Core.Update           (UpId, UpdateVote, UpdateProposal, BlockVersionData)
+import           Pos.Security              (SecurityParams (..))
+import           Pos.Ssc.Message           (MCOpening, MCShares, MCCommitment,
+                                            MCVssCertificate)
+import           Pos.Util.Chrono           (NewestFirst, OldestFirst, NE)
+
+-- | The interface to a diffusion layer, i.e. some component which takes care
+-- of getting data in from and pushing data out to a network.
+data Logic m = Logic
```

Comment:
Why is it called `Logic` if it's used as csl interface for diffusion layer? Why didn't you decide to go with set of constraints instead? 
I do not understand design decisions of this PR tbh, I don't find PR description deep enough, it's more stating things rather than explaining them.

Reply:
Oops, the comment is totally wrong. I must have copy/pasted the diffusion file and then re-wrote it to have the logic stuff.

Reply:
> Why didn't you decide to go with set of constraints instead?

Using typeclass constraints could get us to a similar solution, but it wouldn't save us any work nor simplify anything. Ultimately we would have to discharge these constraints, and that amounts to giving exactly what we have here: values of type `Logic` and `Diffusion` or some union of the two.